appserver/common/container-common/src/main/java/com/sun/enterprise/container/common/impl/EntityManagerWrapper.java

```diff
@@ -397,8 +397,8 @@ public void remove(Object entity) {
                 callFlowAgent.entityManagerMethodStart(EntityManagerMethod.FIND_CLASS_OBJECT_LOCKMODETYPE_PROPERTIES);
             }
             EntityManager delegate = _getDelegate();
-            returnValue = _getDelegate().find(entityClass, primaryKey, lockMode, properties);
-            clearPersistenceContextPerhaps(delegate);
+            returnValue = delegate.find(entityClass, primaryKey, lockMode, properties);
```

Comment:
agree that should use the delegate obtained 1 statement before instead of calling _getDelegate() again